Add visibility change listener for performance telemetry

konstantin-msft opened the pull request:
- Adds counters to `PerformanceEvent`.
- Adds a method to increment counters to `PerformanceEvent`. 
- Adds a `visibilityChangeCount` to performance data to count Page Visibility changes during the AcquireTokenSilent, AcquireTokenByCode, SsoSilent calls. 
- Tests included.

bmahall committed:
ATS- event visibility change

bmahall committed:
Change files

bmahall committed:
Add visibility change for ssoSilent and ATC

bmahall committed:
add tests

bmahall committed:
test

bmahall committed:
add test for visChange true

bmahall committed:
Update @azure-msal-browser-59aee50c-dfa5-47b3-aba3-935707c03d9f.json

bmahall committed:
Update @azure-msal-common-50a60241-5eeb-4067-803d-bca7c2f9b0bb.json

bmahall committed:
Remove abbreviations

konstantin-msft committed:
- Address review comments.
- Minor code clean up.

konstantin-msft committed:
- Remove redundant change file.

konstantin-msft committed:
Change files

konstantin-msft committed:
Change files

konstantin-msft committed:
- Change files.

konstantin-msft committed:
Change files

konstantin-msft commented:
Copy of https://github.com/AzureAD/microsoft-authentication-library-for-js/pull/5488. All code review comments are addressed.

peterzenz commented on lib/msal-browser/src/app/ClientApplication.ts:
```diff
@@ -415,6 +419,20 @@ export abstract class ClientApplication {
+
+        this.logger.info("Perf: Visibility change detected in ", measurement.event.name);
+        measurement.addStaticFields({
+            visibilityChange: true,
```
I think the visibility change will be really diagnostic but as I'm looking at it, does anyone have any thoughts on whether we should make this a count vs a flag?  For instance, in the course of this call, the visibility changed 3x.

tnorling replied:
I don't have a strong opinion but my thoughts are "why not"

konstantin-msft replied:
I think we can. Do you want me to make a change in this pr or make a follow-up?

peterzenz replied:
I'm thinking of how I use the browser and I often ctrl+click on links which tends to open them in a "hidden" tab.

Let's change it to a counter.  I'm ok with either going in this PR or the next one.  If you want to close this one out as is that's OK with me.

sameerag replied:
I am leaning towards the counter, which gives a more accurate status.

konstantin-msft replied:
Done.

konstantin-msft committed:
- Add counters to `PerformanceEvent`.
- Add method to increment counters to `PerformanceEvent`.
- Make `visibilityChange` a counter instead of a flag.

hectormmg reviewed (approved):
LGTM!

ghost commented:
:tada:`@azure/msal-browser@v2.33.0` has been released which incorporates this pull request.:tada:

Handy links:
* [Release Notes](https://github.com/AzureAD/microsoft-authentication-library-for-js/releases/tag/msal-browser-v2.33.0)
* [Official NPM Release](https://www.npmjs.com/package/@azure/msal-browser/v/2.33.0)
